include/seastar/core/reactor.hh

```diff
@@ -190,7 +190,7 @@ public:
 class reactor {
 private:
     struct task_queue;
-    using task_queue_list = circular_buffer_fixed_capacity<task_queue*, max_scheduling_groups()>;
+    using task_queue_list = circular_buffer_fixed_capacity<task_queue*, 1 << log2ceil(max_scheduling_groups())>;
```

Comment:
Because of this log2ceil, there might be a difference between what max_scheduling_groups returns (e.g., it might be 13) and what task_queue_list.size() returns. 
Beyond being confusing, I think this difference can actually cause create_scheduling_group() to crash if a non-power-of-two max is set by the user: It will then successfully call allocate_scheduling_group_id(), and then fail assert(id < max_scheduling_groups()).
Please try your new unit test with a non-power-of-two length - I suspect it will crash (if not, I wonder how).